### PR TITLE
Chore: Add plan action class to control plan flow

### PR DESCRIPTION
**Pull request by mykalmax**
Previous update to plan flow has introduced some regression. It inaccurately display stages and messages (Button Labels). Also not showing changes to Model Metadata and when there are some backfills but no changes.
I added class to control current `plan-action` from single source.

### CODE REVIEW

File: web/client/src/library/components/plan/PlanActionsDescription.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          </span>
+        )}
+        {no_auto_categorization && (
+          <span className="inline-block mr-1">
+            also set <b>Change Category</b> manually
+          </span>
+        )}
+        {isFalse(isStringEmptyOrNil(restate_models)) && (
+          <span className="inline-block mr-1">
+            and restate folowing models <b>{restate_models}</b>
```
Comment (vchan) [inline web/client/src/library/components/plan/PlanActionsDescription.tsx:62]:
```suggestion
            and restate the following models <b>{restate_models}</b>
```

File: web/client/src/library/components/plan/PlanApplyStageTracker.tsx [hunk 2]
```diff
@@ -90,41 +91,46 @@ export default function PlanApplyStageTracker(): JSX.Element {
           </small>
         </div>
       )}
-      {isFalseOrNil(hasChanges) || isNil(changes) ? (
+      {planAction.isRunning || (isNotNil(changes) && isTrue(hasChanges)) ? (
```
Comment (vchan) [inline web/client/src/library/components/plan/PlanApplyStageTracker.tsx:94]: I know this isn't exactly new, but what's the difference been changes and hasChanges?

File: web/client/src/library/components/plan/PlanOptions.tsx [hunk 3]
```diff
@@ -142,8 +138,16 @@ export default function PlanOptions({
                         info="Ensure that new snapshots have no data gaps when
               comparing to existing snapshots for matching
               models in the target environment"
-                        enabled={Boolean(no_gaps)}
-                        disabled={isInitialPlanRun}
+                        enabled={
```
Comment (vchan) [inline web/client/src/library/components/plan/PlanOptions.tsx:141]: Can InputToggle be enabled and disabled at the same time?

File: web/client/src/library/pages/ide/IDE.tsx [hunk 4]
```diff
@@ -328,17 +327,30 @@ export default function PageIDE(): JSX.Element {
...
 
     setPlanApply(planApply)
   }
 
+  function updateEnviroments(data: Optional<Environments>): void {
+    if (
+      isNil(data) ||
+      isNil(data.environments) ||
+      isObjectEmpty(data) ||
+      isObjectEmpty(data.environments)
```
Comment (vchan) [inline web/client/src/library/pages/ide/IDE.tsx:340]: Can this be condensed to something like isObjectNotEmpty(data?.environments)?

**Review by vchan — APPROVED** [review]

File: web/client/src/library/components/plan/PlanOptions.tsx (hunk 3 above)
Comment (mykalmax) [reply web/client/src/library/components/plan/PlanOptions.tsx:141]: `enabled` is referring to Toggle State but `disabled` is referring to input state, so `enabled` Toggle can be `disabled`. It is poor naming, i will rename it to something like `checked` in separate PR

File: web/client/src/library/components/plan/PlanApplyStageTracker.tsx (hunk 2 above)
Comment (mykalmax) [reply web/client/src/library/components/plan/PlanApplyStageTracker.tsx:94]: it is possible these two don't much, need to refactor it so `hasChanges` is just derived from `changes `

**Pull request merged**